audiobook_generator/book_parsers/epub_book_parser.py

```diff
@@ -0,0 +1,85 @@
+from audiobook_generator.config.general_config import GeneralConfig
+
+logger = logging.getLogger(__name__)
+
+
+class EpubBookParser(BaseBookParser):
+    def __init__(self, config: GeneralConfig):
+        super().__init__(config)
+        self.book = epub.read_epub(self.config.input_file)
+
+    def __str__(self) -> str:
+        return super().__str__()
+
+    def validate_config(self):
+        if self.config.input_file is None:
+            raise ValueError("Epub Parser: Input file cannot be empty")
+        if not self.config.input_file.endswith(".epub"):
+            raise ValueError(f"Epub Parser: Unsupported file format: {self.config.input_file}")
+
+    def get_book(self):
+        return self.book
+
+    def get_book_title(self) -> str:
+        if self.book.get_metadata('DC', 'title'):
+            return self.book.get_metadata("DC", "title")[0][0]
+        return "Untitled"
+
+    def get_book_author(self) -> str:
+        if self.book.get_metadata('DC', 'creator'):
+            return self.book.get_metadata("DC", "creator")[0][0]
+        return "Unknown"
+
+    def get_chapters(self, break_string) -> List[Tuple[str, str]]:
+        chapters = []
+        for item in self.book.get_items_of_type(ebooklib.ITEM_DOCUMENT):
+            content = item.get_content()
+            soup = BeautifulSoup(content, "lxml")
+            title = soup.title.string if soup.title else ""
+            raw = soup.get_text(strip=False)
+            logger.debug(f"Raw text: <{raw[:]}>")
+
+            # Replace excessive whitespaces and newline characters based on the mode
+            if self.config.newline_mode == "single":
+                cleaned_text = re.sub(r"[\n]+", break_string, raw.strip())
+            elif self.config.newline_mode == "double":
+                cleaned_text = re.sub(r"[\n]{2,}", break_string, raw.strip())
+            else:
+                raise ValueError(f"Invalid newline mode: {self.config.newline_mode}")
+
+            logger.debug(f"Cleaned text step 1: <{cleaned_text[:]}>")
+            cleaned_text = re.sub(r"\s+", " ", cleaned_text)
+            logger.debug(f"Cleaned text step 2: <{cleaned_text[:100]}>")
+
+            # Removes end-note numbers
+            if self.config.remove_endnotes:
+                cleaned_text = re.sub(r'(?<=[a-zA-Z.,!?;”")])\d+', "", cleaned_text)
+                logger.debug(f"Cleaned text step 4: <{cleaned_text[:100]}>")
+
+            # fill in the title if it's missing
+            if not title:
+                title = cleaned_text[:60]
+            logger.debug(f"Raw title: <{title}>")
+            title = self._sanitize_title(break_string)
+            logger.debug(f"Sanitized title: <{title}>")
+
+            chapters.append((title, cleaned_text))
+            soup.decompose()
+        return chapters
+
+    def _sanitize_title(self, break_string) -> str:
+        # replace MAGIC_BREAK_STRING with a blank space
+        # strip incase leading bank is missing
```

Comment:
We may need a `break_string.strip()` here because sometimes the text happens to be split into two chunks and breaks the break_string. Like ` @BRK#` been split into ` ` and `@BRK#` then the replace won't work.

https://github.com/p0n1/epub_to_audiobook/blob/f7fa674ef1e0f36660b87c0f5275d484498d0e9e/epub_to_audiobook.py#L267

Reply:
had to revert this
as title become: `T_h_e_L_i_f_e_a_n_d_A_d_v_e_n_t_u_r_e_s_o_f_R_o_b_i_n_s_o_n_C_r_u_s_o_e`
instead of: `The_Life_and_Adventures_of_Robinson_Crusoe`